hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -50,10 +52,16 @@ public byte[] getPrefix() {
 
   @Override
   public boolean filterRowKey(Cell firstRowCell) {
-    if (firstRowCell == null || this.prefix == null) return true;
-    if (filterAllRemaining()) return true;
+    if (firstRowCell == null || this.prefix == null) {
+      return true;
+    }
+    if (filterAllRemaining()) {
+      return true;
+    }
     int length = firstRowCell.getRowLength();
-    if (length < prefix.length) return true;
```

Comment:
These are actually not strictly required to resolve the ticket. Just the checkstyle report contained these warnings:
```
(blocks) NeedBraces: 'if' construct must use '{}'s.
(blocks) NeedBraces: 'if' construct must use '{}'s.
(blocks) NeedBraces: 'if' construct must use '{}'s.
(blocks) NeedBraces: 'if' construct must use '{}'s.
```

Reply:
Actually this if looks buggy.
If the firstRowCell is lexicographically smaller than the prefix, then we still want to provide the hint, regardless of the key length.

We can probably just drop this, I believe that the compare later will also handle this case.
This was supposed to be an optization, but it is not relevant when we provide hints, and in fact hurts performance.

Please fix and also add test cases for this.

Reply:
Thanks, removed this if statement and added a unit test.